--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -69,7 +69,11 @@ function(PINOCCHIO_TARGET target_name)
   set(LIB_NAME "${target_name}")

   # Manage different scope type if building an interface or shared library
-  set(LIBRARY_TYPE SHARED)
+  if(BUILD_SHARED_LIBS)
+    set(LIBRARY_TYPE SHARED)
+  else()
+    set(LIBRARY_TYPE STATIC)
+  endif()
   set(LIBRARY_PUBLIC_SCOPE PUBLIC)
   if(ARGS_INTERFACE)
     set(LIBRARY_TYPE INTERFACE)
